app/hooks/useOrders.tsx

```diff
@@ -59,17 +60,47 @@ export function intoOrders(response: AxiosResponse<Entity>): Order[] {
     });
 }
 
+async function checkIfOrderWasMatched(
+  prevOrders: Order[],
+  orders: Order[]
+): Promise<Order[]> {
+  if (!prevOrders) {
+    return orders;
+  }
+
+  try {
+    for (const prevOrder of prevOrders) {
+      const order = orders.find(o => o.id === prevOrder.id);
+
+      if (order && order.state.open.value !== prevOrder.state.open.value) {
+        await mutate('/swaps');
```

Comment:
Will mutating `/swaps` refresh all swaps?

Polling `/swaps` is actually no issue, that is cheap because it just hits the DB. It is more the individual swaps that are a concern.